### PR TITLE
DEVDOCS-3443-add-external_merchant_id

### PR DESCRIPTION
**Pull request by traci707**
[DEVDOCS-3443](https://jira.bigcommerce.com/browse/DEVDOCS-3443)
Added the `external_merchant_id` field.

### CODE REVIEW

**Review by bc-tgomez — APPROVED** [review]
LGTM!

File: reference/orders.v2.oas2.yml [hunk 1]
```diff
@@ -4091,7 +4091,13 @@ components:
...
-          description: 'ID of the order in another system. For example, the Amazon Order ID if this is an Amazon order.This field can be updated in a /POST, but using a /PUT to update the order will return a 400 error. The field ''external_id'' cannot be written to. Please remove it from your request before trying again. It can not be overwritten once set.'
+          description: 'ID of the order in another system. For example, the Amazon Order ID if this is an Amazon order.This field can be updated in a /POST, but using a /PUT to update the order will return a 400 error. The field `external_id` cannot be written to. Please remove it from your request before trying again. It can not be overwritten once set.'
+          example: null
+          oneOf:
+            - type: string
+            - type: null
+        external_merchant_id:
+          description: ID of the merchant in the other system. For example, the Amazon Order ID or Facebook page ID can be updated in a /POST, but using a /PUT to update the order will return a 400 error. You can not write to or update the `external_merchant_id` after it has been set. Please remove it from your request before trying again.
```
Comment (bc-vincent-zhao) [inline reference/orders.v2.oas2.yml:4100]: https://threads.com/34377770087

Should we remove the `Amazon Order ID` and just leave the `Facebook page ID` ?

Comment (bc-vincent-zhao) [reply reference/orders.v2.oas2.yml:4100]: 🍹 : I'm not sure if it worth while to point out that both `external_id` and `external_merchant_id` are currently meant to represent "upstream" orders from external system, not downstream, as they only refers to the source of truth for orders.

In a complicated setup Amazon -> BC -> Merchant ERP, Amazon is the source of truth for orders and those 2 fields are used to describe the link between Amazon -> BC, it should NOT be used to describe BC -> Merchant ERP, we currently ask merchants to use Metafields for that.

Comment (traci707) [reply reference/orders.v2.oas2.yml]: @bc-vincent-zhao thank you for your comments. I made some changes but did not add the example. Let me know if I should add the example.

**Review by bc-vincent-zhao — APPROVED** [review]
@traci707 thanks for the change, looks good 👍

**Pull request merged**